src/WebJobs.Script.WebHost/Management/LinuxSpecialization/PackageDownloadHandler.cs

```diff
@@ -214,5 +229,57 @@ private async Task<bool> IsResourceAccessibleWithoutAuthorization(string resourc
                 return false;
             }
         }
+
+        private string CopyPackageFile(IFileSystem fileSystem = null)
+        {
+            fileSystem ??= FileUtility.Instance;
+            var packageFolderPath = _environment.GetSitePackagesPath();
+
+            if (!fileSystem.Directory.Exists(packageFolderPath))
+            {
+                CopyPackageFileFailed($"{nameof(CopyPackageFile)} failed. {ScriptConstants.SitePackagesFolderName} folder doesn't exist at {packageFolderPath}.");
+            }
+
+            var packageNameTxtPath = _environment.GetSitePackageNameTxtPath();
+            if (!fileSystem.File.Exists(packageNameTxtPath))
+            {
+                CopyPackageFileFailed($"{nameof(CopyPackageFile)} failed. {ScriptConstants.SitePackageNameTxtFileName} doesn't exist at {packageNameTxtPath}.");
+            }
+
+            var packageFileName = fileSystem.File.ReadAllText(packageNameTxtPath);
+
+            if (string.IsNullOrEmpty(packageFileName))
+            {
+                CopyPackageFileFailed($"{nameof(CopyPackageFile)} failed. {ScriptConstants.SitePackageNameTxtFileName} is empty at {packageNameTxtPath}.");
+            }
+
+            var packageFilePath = fileSystem.Path.Combine(packageFolderPath, packageFileName);
+            if (!fileSystem.File.Exists(packageFilePath))
+            {
+                CopyPackageFileFailed($"{nameof(CopyPackageFile)} failed. {packageFileName} doesn't exist at {packageFilePath}.");
+            }
+
+            var packageFileInfo = fileSystem.FileInfo.FromFileName(packageFilePath);
+            if (packageFileInfo.Length == 0)
+            {
+                CopyPackageFileFailed($"{nameof(CopyPackageFile)} failed. {packageFileName} size is zero at {packageFilePath}.");
+            }
+
+            var tmpPath = fileSystem.Path.GetTempPath();
```

Comment:
lets download to EnvironmentSettingNames.LocalSitePackages instead of temp to denote the intent.

Reply:
i think this is fine for now.